### PR TITLE
*: reduce raft cluster lock and replace some place with atomic

### LINKED ISSUE #8785: *: split raft cluster lock

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close #8785

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
### Release note

```release-note
None.
```

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -2319,25 +2318,19 @@ func (c *RaftCluster) loadMinResolvedTS() {
...
 	if !c.isInitialized() {
 		return math.MaxUint64
 	}
-	return c.minResolvedTS
+	return c.minResolvedTS.Get()
 }
 
 // GetStoreMinResolvedTS returns the min resolved ts of the store.
 func (c *RaftCluster) GetStoreMinResolvedTS(storeID uint64) uint64 {
-	c.RLock()
```
Comment (lhy1024) [inline server/cluster/cluster.go:2339]: it is not necessary

**Review by okJiang — COMMENTED** [review]
Why not use atomic.Value?

**Comment by lhy1024** [comment]
> Why not use atomic.Value?

Sure, at first I thought it was possible that we'd need to manually lock and unlock to control more of the code, but now it seems that atomic is sufficient for these three places

**Review by JmPotato — APPROVED** [review]

**Pull request merged**